cpp/src/arrow/util/thread_pool.h

```diff
@@ -373,7 +374,7 @@ class ARROW_EXPORT ThreadPool : public Executor {
   State* state_;
   bool shutdown_on_destroy_;
 #ifndef _WIN32
-  pid_t pid_;
+  std::atomic<pid_t> pid_;
```

Comment:
This gives an uninitialized random value.
Zero initialize with `std::atomic<pid_t> pid_{}`.

Reply:
It's initialized at https://github.com/apache/arrow/blob/master/cpp/src/arrow/util/thread_pool.cc#L225. And I don't think it should be zero when it's initialized.